<center> <h1>Universidad Nacional de San Agustín de Arequipa</h1> </center> 
<center> <h1>Escuela Profesional de Ingeniería de Telecomunicaciones</h1> </center> 

<center> <h1> </h1> </center> 

<center><img src="https://www.unsa.edu.pe/wp-content/uploads/sites/3/2018/05/Logo-UNSA.png" width="380" height="4200"></center>


<center> <h1>Curso : Computación 1</h1> </center> 

![linea 1](https://user-images.githubusercontent.com/19308295/115926252-2b8a0c00-a448-11eb-9d9c-b43beaf0ff68.png)

<center> <h2>GRUPO "C" - Nº2</h2> </center> 

<center> <h2>Ingeniero Renzo Bolivar - Docente DAIE</h2> </center>

<h2>Intengrantes:  </h2>
<h2>    

    💫 LARICO YUCRA, CRISTHIAN DENIS. 
    💫 CHALLCO ANCCASI, CESAR AUGUSTO. 
    💫 MAMANI ALVAREZ, RUDY.
    💫 PRIETO TITO, MANUEL ISMAEL.
    💫 LLERENA PASTOR, JHORDAN ALEXANDER.

</h2>

![linea 1](https://user-images.githubusercontent.com/19308295/115926252-2b8a0c00-a448-11eb-9d9c-b43beaf0ff68.png)

<center> <h1>INVESTIGACIÓN FORMATIVA</h1> </center> 
<center> <h1>PROYECTO FINAL</h1> </center> 
<center> <h1>PYTHON - Inteligencia Artificial</h1> </center> 

![linea 1](https://user-images.githubusercontent.com/19308295/115926252-2b8a0c00-a448-11eb-9d9c-b43beaf0ff68.png)

## OBJETIVOS

Los Objetivos de la investigación formativa son:

- **Competencia Comunicativa** Presentación de sus resultados con lenguaje de programación Python utilizando los archivos Jupyter Notebook.
- **Competencia Aprendizaje**: con las aptitudes en **Descomposición** (desarticular el problema en pequeñas series de soluciones), **Reconocimiento de Patrones** (encontrar simulitud al momento de resolver problemas), **Abstracción** (omitir información relevante), **Algoritmos** (pasos para resolución de un problema).
- **Competencia de Trabajo en Equipo**: exige habilidades individuales y grupales orientadas a la cooperación, planificación, coordinación, asignación de tareas, cumplimiento de tareas y solución de conflictos en pro de un trabajo colectivo, utilizando los archivos Jupyter Notebook los cuales se sincronizan en el servidor Gitlab con comandos Git.

![linea 1](https://user-images.githubusercontent.com/19308295/115926252-2b8a0c00-a448-11eb-9d9c-b43beaf0ff68.png)

<center> <h1>Aplicación en IA</h1> </center> 
<center> <h1>Sistema Recomendador</h1> </center> 

<div class="alert alert-info">
El Sistema recomendador deberá encontrar la <strong>compatibilidad o similitud</strong> entre un grupo de personas encuestadas, en las áreas de:

</div>

<div class="alert alert-info">
    
   -Musica
   
   -Peliculas
    
   -Comida
</div>

<div class="alert alert-info">

    
   La <strong>compatibilidad o similitud</strong> será encontrada con el algoritmo de <strong>Correlación de Pearson</strong> y será verificada con la <strong>La Matrix de Correlación de Pearson de la librería Numpy</strong>
    
</div>

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

<center> <h1>  🥩 COMIDAS PERUANAS 🇵🇪 </h1> </center>

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

## 1.- Dataset

<div class="alert alert-info">

    
   Para poder tener el  <strong>Datset</strong> hay que recolectar información a travez de encuentas.
    
</div>

## 🏁 NUESTRA ENCUESTA 

La encuesta la realizamos en Google-Form donde la persona a encuestar nos dara una opinion acerca de los gustos que tiene sobre la **COMIDAS PERUANA** .
Este respondera en una escala de 1 a 5 donde  :
- Si escoge 1, no le  gusta la comida propuesta.
- Si escoge 5, si el encanta la comida propuesta.  

Todo esto estara dado en la **Escala de liker**.

## 🏁 Capturas de Nuestro Formulario de Google

<a href="https://ibb.co/72x3j4c"><img src="https://i.ibb.co/sypTbHS/Whats-App-Image-2021-12-14-at-12-26-27-AM-1.jpg" alt="Whats-App-Image-2021-12-14-at-12-26-27-AM-1" border="0"></a>

<a href="https://ibb.co/zSjZ87z"><img src="https://i.ibb.co/MRW78Gm/Whats-App-Image-2021-12-14-at-12-26-27-AM-2.jpg" alt="Whats-App-Image-2021-12-14-at-12-26-27-AM-2" border="0"></a>

<a href="https://ibb.co/3YMzDHN"><img src="https://i.ibb.co/5xnFNV8/Whats-App-Image-2021-12-14-at-12-26-27-AM.jpg" alt="Whats-App-Image-2021-12-14-at-12-26-27-AM" border="0"></a><br /><a target='_blank' href='https://es.imgbb.com/'>subir fotos a google</a><br />

In [1]:
import numpy as np
import pandas as pd

In [2]:
data = pd.read_csv ('Comidas - Respuestas del  formulario.csv')
data

Correo electrónico  Seco de cabrito  Ocopa Arequipeña  \
0               cchallcoa@unsa.edu.pe                4                 3   
1              jllerenapa@unsa.edu.pe                3                 4   
2                mprietot@unsa.edu.pe                3                 4   
3                ruizcruza9@gmail.com                4                 4   
4             alessccorahua@gmail.com                2                 2   
5                aapazatt@unsa.edu.pe                2                 3   
6              rapazaquis@unsa.edu.pe                5                 3   
7               obedcilas15@gmail.com                2                 3   
8                 kathielf3@gmail.com                3                 2   
9       veronica.aracelly17@gmail.com                1                 3   
10                fllacho@unsa.edu.pe                4                 4   
11         gracia.alvarez@ucsp.edu.pe                3                 4   
12                bvilcas@unsa.edu.pe                2                 4   
13              sbarriosr@unsa.edu.pe                3                 4   
14                mchecca@unsa.edu.pe                5                 1   
15         joselynmachaca@upeu.edu.pe                4                 4   
16  rebecacondorizapana2005@gmail.com                1                 2   
17            irodriguezc@unsa.edu.pe                4                 3   
18              gmartinez@unsa.edu.pe                5                 2   
19                kmejiah@unsa.edu.pe                2                 4   
20            matajudios123@gmail.com                5                 3   
21          ivanrodrigo1926@gmail.com                5                 5   
22        madelyquispe793@unsa.edu.pe                2                 3   
23               msuclles@unsa.edu.pe                4                 3   
24             jchambisur@unsa.edu.pe                5                 3   
25               mttitocr@unsa.edu.pe                4                 5   
26               raraujoa@unsa.edu.pe                1                 2   
27              chemystry03@gmail.com                3                 5   
28            hendersonwido@gmail.com                3                 4   
29                 zamyma01@gmail.com                4                 1   

    Carapulcra   Causa Rellena  Papa Rellena  Lomo Saltado  Arroz con Pollo   \
0             4              3             2             4                 2   
1             3              4             2             3                 5   
2             3              5             1             3                 2   
3             3              4             2             5                 2   
4             3              1             3             4                 3   
5             4              3             2             4                 3   
6             4              3             2             3                 4   
7             3              4             3             5                 3   
8             4              2             1             3                 4   
9             2              4             4             3                 1   
10            3              2             3             5                 3   
11            3              2             3             2                 1   
12            2              4             4             3                 4   
13            1              4             3             5                 2   
14            5              5             3             4                 2   
15            3              5             3             4                 2   
16            4              2             3             4                 2   
17            5              3             2             4                 3   
18            3              4             2             4                 5   
19            2              3             3             4    

In [3]:
data.dtypes

Correo electrónico    object
Seco de cabrito        int64
Ocopa Arequipeña       int64
Carapulcra             int64
Causa Rellena          int64
Papa Rellena           int64
Lomo Saltado           int64
Arroz con Pollo        int64
Juane                  int64
Pollo a la Brasa       int64
Pollo a la olla        int64
Estofado de Res        int64
Adobo                  int64
Mondonguito            int64
Rocoto Relleno         int64
Leche de Tigre         int64
Arroz con Mariscos     int64
Pachamanca             int64
Costillar Dorado       int64
Lenteja                int64
Caldo Blanco           int64
dtype: object

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

## 2.-  Distancia Euclidiana

__La distancia euclidiana__ es la generalización del __`teorema de Pitágoras`__.

$$d_{E}(P,Q)=\sqrt{(p_{1}-q_{1})^{2}+(p_{2}-q_{2})^{2}+...+(p_{n}-q_{n})^{2}}=\sqrt{\sum_{i=1}^{n}}(p_{i}-q_{i})^{2})$$

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

## 3.- Correlación de Pearson  (Similitud)

El __coeficiente de correlación de Pearson__ es una medida de __`dependencia lineal entre dos variables aleatorias cuantitativas`__. 

El coeficiente de correlación puede tomar un __rango de valores de +1 a -1__. Un valor de 0 indica que no hay asociación entre las dos variables. __`Un valor mayor que 0 indica una asociación positiva`__ y __`Un valor menor que 0 indica una asociación negativa`__.

$$s = \frac{1}{1+d_{E}}$$ 

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

In [4]:
from scipy.spatial.distance import euclidean

In [5]:
n = data[data.columns[1:]].to_numpy()
m = data[data.columns[0]].to_numpy()
print(n)
print(m)

[[4 3 4 3 2 4 2 3 4 4 5 3 4 3 5 3 4 2 2 3]
 [3 4 3 4 2 3 5 2 2 3 2 3 4 5 2 4 3 2 4 3]
 [3 4 3 5 1 3 2 1 3 2 5 2 3 4 3 5 4 5 4 3]
 [4 4 3 4 2 5 2 1 4 3 5 3 4 2 4 2 4 2 1 4]
 [2 2 3 1 3 4 3 2 4 1 4 2 3 2 5 2 4 1 2 1]
 [2 3 4 3 2 4 3 4 3 2 4 3 2 3 4 4 3 2 4 1]
 [5 3 4 3 2 3 4 2 1 1 4 3 3 4 2 4 3 2 4 2]
 [2 3 3 4 3 5 3 2 4 1 4 3 4 2 3 3 4 2 2 3]
 [3 2 4 2 1 3 4 2 1 3 3 2 4 2 3 3 2 5 4 3]
 [1 3 2 4 4 3 1 1 4 3 5 4 4 2 3 2 4 2 3 5]
 [4 4 3 2 3 5 3 2 3 2 4 3 4 3 4 3 4 4 3 2]
 [3 4 3 2 3 2 1 3 4 2 4 3 2 3 2 4 3 4 3 2]
 [2 4 2 4 4 3 4 2 4 3 5 4 3 3 2 3 4 3 3 3]
 [3 4 1 4 3 5 2 4 1 2 4 3 4 2 4 5 4 5 4 4]
 [5 1 5 5 3 4 2 3 4 4 2 3 4 3 5 2 1 2 3 4]
 [4 4 3 5 3 4 2 2 3 4 3 3 4 2 5 4 3 4 4 3]
 [1 2 4 2 3 4 2 3 4 3 4 3 5 4 3 4 2 3 2 2]
 [4 3 5 3 2 4 3 4 2 3 4 3 4 3 3 4 3 2 3 3]
 [5 2 3 4 2 4 5 3 2 1 2 3 4 5 4 3 2 5 4 3]
 [2 4 2 3 3 4 3 2 3 4 2 3 4 3 5 3 2 4 3 4]
 [5 3 5 2 3 4 2 1 3 2 3 2 4 3 4 4 5 3 2 3]
 [5 5 3 4 3 4 4 4 3 5 2 3 4 3 2 4 3 4 4 3]
 [2 3 2 3 2 4 2 1 2 3 4 2 3 4 5 3 2 3 2 4]
 [4 3 4 4 3

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

## 🏁 Hallando la distancia y similitud  de los encuestados.

In [6]:
from pandas import DataFrame
vector=len(m)
matriz= np.zeros((vector,vector))

for i in range(vector):
    for j in range(vector):
        if j and i== np.NaN:
            continue
        else:
            dist=np.sqrt(np.nansum((n[i]-n[j])**2))
            simil=(1/(1+dist))
            matriz[i][j] = simil

da = pd.DataFrame(matriz,columns=m,index=m)
da

cchallcoa@unsa.edu.pe  \
cchallcoa@unsa.edu.pe                           1.000000   
jllerenapa@unsa.edu.pe                          0.126132   
mprietot@unsa.edu.pe                            0.133677   
ruizcruza9@gmail.com                            0.210897   
alessccorahua@gmail.com                         0.150221   
aapazatt@unsa.edu.pe                            0.161390   
rapazaquis@unsa.edu.pe                          0.131006   
obedcilas15@gmail.com                           0.166667   
kathielf3@gmail.com                             0.127297   
veronica.aracelly17@gmail.com                   0.141188   
fllacho@unsa.edu.pe                             0.182744   
gracia.alvarez@ucsp.edu.pe                      0.142857   
bvilcas@unsa.edu.pe                             0.146392   
sbarriosr@unsa.edu.pe                           0.125000   
mchecca@unsa.edu.pe                             0.150221   
joselynmachaca@upeu.edu.pe                      0.169521   
rebecacondorizapana2005@gmail.com               0.161390   
irodriguezc@unsa.edu.pe                         0.200000   
gmartinez@unsa.edu.pe                           0.116961   
kmejiah@unsa.edu.pe                             0.150221   
matajudios123@gmail.com                         0.175734   
ivanrodrigo1926@gmail.com                       0.132320   
madelyquispe793@unsa.edu.pe                     0.161390   
msuclles@unsa.edu.pe                            0.128496   
jchambisur@unsa.edu.pe                          0.139579   
mttitocr@unsa.edu.pe                            0.111111   
raraujoa@unsa.edu.pe                            0.139579   
chemystry03@gmail.com                           0.135078   
hendersonwido@gmail.com                         0.158945   
zamyma01@gmail.com                              0.141188   

                                   jllerenapa@unsa.edu.pe  \
cchallcoa@unsa.edu.pe                            0.126132   
jllerenapa@unsa.edu.pe                           1.000000   
mprietot@unsa.edu.pe                             0.139579   
ruizcruza9@gmail.com                             0.117878   
alessccorahua@gmail.com                          0.109601   
aapazatt@unsa.edu.pe                             0.144591   
rapazaquis@unsa.edu.pe                           0.182744   
obedcilas15@gmail.com                            0.138026   
kathielf3@gmail.com                              0.144591   
veronica.aracelly17@gmail.com                    0.113504   
fllacho@unsa.edu.pe                              0.142857   
gracia.alvarez@ucsp.edu.pe                       0.128496   
bvilcas@unsa.edu.pe                              0.154387   
sbarriosr@unsa.edu.pe                            0.120771   
mchecca@unsa.edu.pe                              0.116066   
joselynmachaca@upeu.edu.pe                       0.139579   
rebecacondorizapana2005@gmail.com                0.132320   
irodriguezc@unsa.edu.pe                          0.163961   
gmartinez@unsa.edu.pe                            0.156613   
kmejiah@unsa.edu.pe                              0.150221   
matajudios123@gmail.com                          0.128496   
ivanrodrigo1926@gmail.com                        0.166667   
madelyquispe793@unsa.edu.pe                      0.138026   
msuclles@unsa.edu.pe                             0.163961   
jchambisur@unsa.edu.pe                           0.150221   
mttitocr@unsa.edu.pe                             0.128496   
raraujoa@unsa.edu.pe                             0.108152   
chemystry03@gmail.com                            0.179129   
hendersonwido@gmail.com                          0.142857   
zamyma01@gmail.com                               0.122829   

                                   mprietot@unsa.edu.pe  ruizcruza9@gmail.com  \
cchallcoa@unsa.edu.pe                          0.133677              0.210897   
jllerenapa@unsa.edu.pe                         0.139579              0.117878   
mprietot@unsa.edu.pe                   

In [12]:
cl=da.unstack()
cl.sort_values(ascending=False)[range(29,36)]

cchallcoa@unsa.edu.pe        cchallcoa@unsa.edu.pe          1.000000
hendersonwido@gmail.com      aapazatt@unsa.edu.pe           0.231662
aapazatt@unsa.edu.pe         hendersonwido@gmail.com        0.231662
ruizcruza9@gmail.com         cchallcoa@unsa.edu.pe          0.210897
cchallcoa@unsa.edu.pe        ruizcruza9@gmail.com           0.210897
kmejiah@unsa.edu.pe          madelyquispe793@unsa.edu.pe    0.205213
madelyquispe793@unsa.edu.pe  kmejiah@unsa.edu.pe            0.205213
dtype: float64

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

### 🏁 Validación - Matrix de Correlación

Se realiza la validación de los resultados obtenidos con la   `Matriz de Correlación` en `Numpy` 

In [8]:
nampa_matriz=np.corrcoef(da)
print(nampa_matriz)

[[ 1.00000000e+00 -7.95827220e-02 -4.84618849e-02  1.37099052e-01
   6.78581459e-03  9.35146653e-03 -6.13100392e-02  2.00001878e-02
  -6.37490456e-02 -2.04620385e-02  4.95777455e-02 -3.41608668e-02
  -3.29556690e-02 -6.25861963e-02  1.01512504e-02  2.46153907e-02
   1.77448490e-02  8.57090397e-02 -8.27434754e-02 -1.82258325e-02
   5.06101727e-02 -5.36486448e-02  1.42143138e-02 -6.53280517e-02
  -3.68671399e-02 -8.41443293e-02 -2.66199055e-02 -5.85029692e-02
   7.30443088e-03 -2.55663664e-02]
 [-7.95827220e-02  1.00000000e+00 -2.34498579e-02 -8.66382191e-02
  -9.12745499e-02 -2.52150914e-02  7.80701359e-02 -5.38641059e-02
  -8.67029022e-03 -8.64663154e-02 -4.47750908e-02 -5.82038202e-02
  -4.63609198e-03 -6.18043344e-02 -6.51639555e-02 -4.06830445e-02
  -4.93867379e-02  1.23220329e-02  2.92761452e-02 -1.41409703e-02
  -6.00941668e-02  4.56648875e-02 -4.10414057e-02  3.93955659e-02
   4.06150940e-03 -2.67840690e-02 -9.98937636e-02  6.40526976e-02
  -2.35694005e-02 -5.63846598e-02]
 [-4.8

In [9]:
correlacion=pd.DataFrame(nampa_matriz,columns=m,index=m)
correlacion.round(decimals=4)

cchallcoa@unsa.edu.pe  \
cchallcoa@unsa.edu.pe                             1.0000   
jllerenapa@unsa.edu.pe                           -0.0796   
mprietot@unsa.edu.pe                             -0.0485   
ruizcruza9@gmail.com                              0.1371   
alessccorahua@gmail.com                           0.0068   
aapazatt@unsa.edu.pe                              0.0094   
rapazaquis@unsa.edu.pe                           -0.0613   
obedcilas15@gmail.com                             0.0200   
kathielf3@gmail.com                              -0.0637   
veronica.aracelly17@gmail.com                    -0.0205   
fllacho@unsa.edu.pe                               0.0496   
gracia.alvarez@ucsp.edu.pe                       -0.0342   
bvilcas@unsa.edu.pe                              -0.0330   
sbarriosr@unsa.edu.pe                            -0.0626   
mchecca@unsa.edu.pe                               0.0102   
joselynmachaca@upeu.edu.pe                        0.0246   
rebecacondorizapana2005@gmail.com                 0.0177   
irodriguezc@unsa.edu.pe                           0.0857   
gmartinez@unsa.edu.pe                            -0.0827   
kmejiah@unsa.edu.pe                              -0.0182   
matajudios123@gmail.com                           0.0506   
ivanrodrigo1926@gmail.com                        -0.0536   
madelyquispe793@unsa.edu.pe                       0.0142   
msuclles@unsa.edu.pe                             -0.0653   
jchambisur@unsa.edu.pe                           -0.0369   
mttitocr@unsa.edu.pe                             -0.0841   
raraujoa@unsa.edu.pe                             -0.0266   
chemystry03@gmail.com                            -0.0585   
hendersonwido@gmail.com                           0.0073   
zamyma01@gmail.com                               -0.0256   

                                   jllerenapa@unsa.edu.pe  \
cchallcoa@unsa.edu.pe                             -0.0796   
jllerenapa@unsa.edu.pe                             1.0000   
mprietot@unsa.edu.pe                              -0.0234   
ruizcruza9@gmail.com                              -0.0866   
alessccorahua@gmail.com                           -0.0913   
aapazatt@unsa.edu.pe                              -0.0252   
rapazaquis@unsa.edu.pe                             0.0781   
obedcilas15@gmail.com                             -0.0539   
kathielf3@gmail.com                               -0.0087   
veronica.aracelly17@gmail.com                     -0.0865   
fllacho@unsa.edu.pe                               -0.0448   
gracia.alvarez@ucsp.edu.pe                        -0.0582   
bvilcas@unsa.edu.pe                               -0.0046   
sbarriosr@unsa.edu.pe                             -0.0618   
mchecca@unsa.edu.pe                               -0.0652   
joselynmachaca@upeu.edu.pe                        -0.0407   
rebecacondorizapana2005@gmail.com                 -0.0494   
irodriguezc@unsa.edu.pe                            0.0123   
gmartinez@unsa.edu.pe                              0.0293   
kmejiah@unsa.edu.pe                               -0.0141   
matajudios123@gmail.com                           -0.0601   
ivanrodrigo1926@gmail.com                          0.0457   
madelyquispe793@unsa.edu.pe                       -0.0410   
msuclles@unsa.edu.pe                               0.0394   
jchambisur@unsa.edu.pe                             0.0041   
mttitocr@unsa.edu.pe                              -0.0268   
raraujoa@unsa.edu.pe                              -0.0999   
chemystry03@gmail.com                              0.0641   
hendersonwido@gmail.com                           -0.0236   
zamyma01@gmail.com                                -0.0564   

                                   mprietot@unsa.edu.pe  ruizcruza9@gmail.com  \
cchallcoa@unsa.edu.pe                           -0.0485                0.1371   
jllerenapa@unsa.edu.pe                          -0.0234               -0.0866   
mprietot@unsa.edu.pe                   

In [11]:
cl=correlacion.abs().unstack()
cl.sort_values(ascending=False)[range(29,36)]

cchallcoa@unsa.edu.pe        cchallcoa@unsa.edu.pe          1.000000
hendersonwido@gmail.com      aapazatt@unsa.edu.pe           0.184527
aapazatt@unsa.edu.pe         hendersonwido@gmail.com        0.184527
cchallcoa@unsa.edu.pe        ruizcruza9@gmail.com           0.137099
ruizcruza9@gmail.com         cchallcoa@unsa.edu.pe          0.137099
kmejiah@unsa.edu.pe          madelyquispe793@unsa.edu.pe    0.121538
madelyquispe793@unsa.edu.pe  kmejiah@unsa.edu.pe            0.121538
dtype: float64

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

## 🏁 RESULTADOS 

Los resultados de similitud obtenidos en **Comidas del Perú** según la tabla de **Correlación** con los siguientes encuestados:

 1. _hendersonwido@gmail.com _ y _aapazatt@unsa.edu.pe_  obtienen el **PRIMER** indice mas alto de similitud de 0.231662
 
 2. _ruizcruza9@gmail.com_ y _cchallcoa@unsa.edu.pe _ obtienen el **SEGUNDO** indice mas alto de similitud 0.210897

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

## 🏁 CONCLUSIONES

## 📌 Podemos concluir que con este proyecto de Inteligencia Artificial, se pudo poner a prueba los conocimientos adquiridos como el aprendizaje del lenguaje PYTHON, así también como el manejo de la plataforma GITLAB para evidenciar cada avance y todo el manejo desde el sistema operativo LINUX.


## 📌 Para el desarrollo del presente proyecto se hizo uso de jupyter notebook y de librerias como numpy y pandas, también se hizo uso spyder para poder crear la pagina en flask y poder plasmar los resultados finales del proyecto de inteligencia articial.

## 📌 La participación, planificación y el trabajo en equipo fue determinante para poder ver avances del proyecto donde cada integrante aporto con sus conocimientos y aptitudes.

![linea 2](https://user-images.githubusercontent.com/19308295/115926262-2fb62980-a448-11eb-8189-c2f10e499944.png)

![linea 1](https://user-images.githubusercontent.com/19308295/115926252-2b8a0c00-a448-11eb-9d9c-b43beaf0ff68.png)

## Referencias

- __analyticslane__:

### Inicialización de arrays con ceros con np.zeros() https://www.analyticslane.com/2019/10/09/numpy-basico-inicializacion-de-arrays-en-numpy/

- __stackoverflow__: 
### Los pares de correlación más altos de una matriz de correlación .unstack( ) https://stackoverflow.com/questions/17778394/list-highest-correlation-pairs-from-a-large-correlation-matrix-in-pandas

- __Youtube__: 

## 🚩Python y la librería Pandas. https://www.youtube.com/watch?v=aoQtF_-rhTI
## 🚩Valores NaN (nulo) https://www.youtube.com/watch?v=PAU92GUyEt4
## 🚩Valores perdidos en pandas https://www.youtube.com/watch?v=fCMrO_VzeL8


- __pandas__: 
### pandas.DataFrame.sort_values https://pandas.pydata.org/pandas-docs/version/0.23/generated/pandas.DataFrame.sort_values.html

![linea 1](https://user-images.githubusercontent.com/19308295/115926252-2b8a0c00-a448-11eb-9d9c-b43beaf0ff68.png)